### PR TITLE
refactor: add ESLint plugin to sort imports

### PR DESCRIPTION
**Pull request by vursen**
Adds the `simple-import-sort` ESLint plugin and sorts imports in the `frontend/demo` folder.

### CODE REVIEW

File: frontend/demo/component/textarea/text-area-basic-features.ts [hunk 1]
```diff
@@ -1,11 +1,10 @@
 import 'Frontend/demo/init'; // hidden-source-line
-
-import { html, LitElement } from 'lit';
-import { customElement } from 'lit/decorators.js';
```
Comment (web-padawan) [inline frontend/demo/component/textarea/text-area-basic-features.ts:4]: nit: could also sort imports below alphabetically (text-area after icons).

File: frontend/demo/component/textfield/text-field-basic-features.ts [hunk 2]
```diff
@@ -1,11 +1,10 @@
 import 'Frontend/demo/init'; // hidden-source-line
-
-import { html, LitElement } from 'lit';
-import { customElement } from 'lit/decorators.js';
```
Comment (web-padawan) [inline frontend/demo/component/textfield/text-field-basic-features.ts:4]: nit: could also sort imports below alphabetically (text-field after icons).

File: frontend/demo/component/timepicker/time-picker-basic-features.ts [hunk 3]
```diff
@@ -1,11 +1,10 @@
 import 'Frontend/demo/init'; // hidden-source-line
-
-import { html, LitElement } from 'lit';
-import { customElement } from 'lit/decorators.js';
```
Comment (web-padawan) [inline frontend/demo/component/timepicker/time-picker-basic-features.ts:4]: nit: could also sort imports below alphabetically (time-picker after icons).

File: frontend/demo/component/tree-grid/tree-grid-rich-content.ts [hunk 4]
```diff
@@ -1,19 +1,18 @@
...
-import type { GridColumnBodyLitRenderer } from '@vaadin/grid/lit.js';
 import '@vaadin/horizontal-layout';
 import '@vaadin/icon';
 import '@vaadin/icons';
 import '@vaadin/vertical-layout';
+import { html, LitElement } from 'lit';
+import { customElement, state } from 'lit/decorators.js';
+import type { GridDataProviderCallback, GridDataProviderParams } from '@vaadin/grid';
+import type { GridColumnBodyLitRenderer } from '@vaadin/grid/lit.js';
+import { columnBodyRenderer } from '@vaadin/grid/lit.js';
```
Comment (web-padawan) [inline frontend/demo/component/tree-grid/tree-grid-rich-content.ts:14]:
```suggestion
import { columnBodyRenderer, type GridColumnBodyLitRenderer } from '@vaadin/grid/lit.js';
```

File: frontend/demo/component/tree-grid/tree-grid-scroll-to-index.ts [hunk 5]
```diff
@@ -1,20 +1,19 @@
 import 'Frontend/demo/init'; // hidden-source-line
-
-import { html, LitElement } from 'lit';
-import { customElement, query, state } from 'lit/decorators.js';
```
Comment (web-padawan) [inline frontend/demo/component/tree-grid/tree-grid-scroll-to-index.ts:4]: nit: could also sort imports below (grid after button).

File: frontend/demo/component/virtuallist/virtual-list-basic.ts [hunk 6]
```diff
@@ -1,16 +1,15 @@
...
 import '@vaadin/horizontal-layout';
 import '@vaadin/vertical-layout';
 import '@vaadin/virtual-list';
-import { virtualListRenderer } from '@vaadin/virtual-list/lit.js';
+import { css, html, LitElement } from 'lit';
+import { customElement, state } from 'lit/decorators.js';
+import { live } from 'lit/directives/live.js';
+import type { Details } from '@vaadin/details';
 import type { VirtualListLitRenderer } from '@vaadin/virtual-list/lit.js';
+import { virtualListRenderer } from '@vaadin/virtual-list/lit.js';
```
Comment (web-padawan) [inline frontend/demo/component/virtuallist/virtual-list-basic.ts:12]: Could combine this with `import type` above.

File: frontend/demo/foundation/lumo-tokens.ts [hunk 7]
```diff
@@ -17,10 +15,12 @@ import '@vaadin/vaadin-lumo-styles/style.js';
 // end::style[]
 // tag::utility-classes[]
 import '@vaadin/vaadin-lumo-styles/utility.js';
+// end::color[]
```
Comment (web-padawan) [inline frontend/demo/foundation/lumo-tokens.ts:18]: This comment seems to be off, please move it to where it was.

File: frontend/demo/foundation/material-tokens.ts [hunk 8]
```diff
@@ -1,10 +1,10 @@
 // Import all Material CSS custom properties into the global style scope
 // tag::color[]
 import '@vaadin/vaadin-material-styles/color';
-// end::color[]
-import { colorDark } from '@vaadin/vaadin-material-styles/color'; // hidden-source-line
 // tag::typography[]
 import '@vaadin/vaadin-material-styles/typography';
+// end::color[]
+import { colorDark } from '@vaadin/vaadin-material-styles/color'; // hidden-source-line
```
Comment (web-padawan) [inline frontend/demo/foundation/material-tokens.ts:7]: Same as in Lumo version, the `end: color` seems to be off.

File: frontend/demo/fusion/authentication/auth.ts [hunk 9]
```diff
@@ -7,10 +7,10 @@ import {
   logout as logoutImpl,
   type LogoutOptions,
 } from '@vaadin/hilla-frontend';
+import type UserInfo from 'Frontend/generated/com/vaadin/demo/fusion/security/authentication/UserInfo';
```
Comment (web-padawan) [inline frontend/demo/fusion/authentication/auth.ts:10]: I thnk this import should be inside of `tag::userinfo[]`.

File: frontend/demo/fusion/forms/field-strategy/my-binder.ts [hunk 10]
```diff
@@ -1,5 +1,9 @@
+import type {
```
Comment (web-padawan) [inline frontend/demo/fusion/forms/field-strategy/my-binder.ts:1]: Could combine this with `Binder` import below into single statement.

**Review by web-padawan — CHANGES_REQUESTED** [review]

File: frontend/demo/component/badge/badge-highlight.ts [hunk 11]
```diff
@@ -1,12 +1,12 @@
...
-import { customElement, state } from 'lit/decorators.js';
 import '@vaadin/grid';
-import { columnBodyRenderer } from '@vaadin/grid/lit.js';
 import '@vaadin/icon';
 import '@vaadin/icons';
+import { html, LitElement } from 'lit';
+import { customElement, state } from 'lit/decorators.js';
+import { columnBodyRenderer } from '@vaadin/grid/lit.js';
+import { getReports, ReportStatus } from 'Frontend/demo/domain/DataService'; // hidden-source-line
+import type Report from 'Frontend/generated/com/vaadin/demo/domain/Report'; // hidden-source-line
```
[the remark this answers is not in the document]
Comment (vursen) [reply frontend/demo/component/badge/badge-highlight.ts:9]: Good point. I think I'll create a separate PR for these changes.

File: frontend/demo/component/badge/badge-icons-only-table.ts [hunk 12]
```diff
@@ -1,13 +1,13 @@
...
-import { columnBodyRenderer } from '@vaadin/grid/lit.js';
-import type { GridColumnBodyLitRenderer } from '@vaadin/grid/lit.js';
 import '@vaadin/icon';
 import '@vaadin/icons';
+import { html, LitElement } from 'lit';
+import { customElement, state } from 'lit/decorators.js';
+import type { GridColumnBodyLitRenderer } from '@vaadin/grid/lit.js';
+import { columnBodyRenderer } from '@vaadin/grid/lit.js';
+import { getUserPermissions } from 'Frontend/demo/domain/DataService'; // hidden-source-line
+import type UserPermissions from 'Frontend/generated/com/vaadin/demo/domain/UserPermissions'; // hidden-source-line
```
[the remark this answers is not in the document]
Comment (vursen) [reply frontend/demo/component/badge/badge-icons-only-table.ts:10]: I'll create a separate PR for these changes.

File: frontend/demo/component/badge/badge-interactive.ts [hunk 13]
```diff
@@ -7,11 +6,12 @@ import '@vaadin/icon';
...
 import '@vaadin/horizontal-layout';
 import '@vaadin/vertical-layout';
-import type { Button } from '@vaadin/button';
-import type { ComboBoxChangeEvent } from '@vaadin/combo-box';
 import { html, LitElement } from 'lit';
 import { customElement, state } from 'lit/decorators.js';
 import { repeat } from 'lit/directives/repeat.js';
+import type { Button } from '@vaadin/button';
+import type { ComboBoxChangeEvent } from '@vaadin/combo-box';
+import { getPeople } from 'Frontend/demo/domain/DataService'; // hidden-source-line
```
[the remark this answers is not in the document]
Comment (vursen) [reply frontend/demo/component/badge/badge-interactive.ts:14]: I'll create a separate PR for these changes.

File: frontend/demo/component/board/example-chart.ts [hunk 14]
```diff
@@ -1,9 +1,9 @@
-import { getViewEvents } from 'Frontend/demo/domain/DataService'; // hidden-source-line
-import type ViewEvent from 'Frontend/generated/com/vaadin/demo/domain/ViewEvent'; // hidden-source-line
+import '@vaadin/charts';
 import { css, html, LitElement } from 'lit';
 import { customElement, state } from 'lit/decorators.js';
 import { repeat } from 'lit/directives/repeat.js';
-import '@vaadin/charts';
+import { getViewEvents } from 'Frontend/demo/domain/DataService'; // hidden-source-line
+import type ViewEvent from 'Frontend/generated/com/vaadin/demo/domain/ViewEvent'; // hidden-source-line
```
[the remark this answers is not in the document]
Comment (vursen) [reply frontend/demo/component/board/example-chart.ts:6]: I'll create a separate PR for these changes.

File: frontend/demo/component/combobox/combo-box-presentation.ts [hunk 15]
```diff
@@ -1,11 +1,10 @@
...
-
+import '@vaadin/combo-box';
 import { html, LitElement } from 'lit';
 import { customElement, state } from 'lit/decorators.js';
-import '@vaadin/combo-box';
-import { comboBoxRenderer } from '@vaadin/combo-box/lit.js';
-import type { ComboBoxLitRenderer } from '@vaadin/combo-box/lit.js';
 import type { ComboBoxFilterChangedEvent } from '@vaadin/combo-box';
+import type { ComboBoxLitRenderer } from '@vaadin/combo-box/lit.js';
+import { comboBoxRenderer } from '@vaadin/combo-box/lit.js';
```
[the remark this answers is not in the document]
Comment (vursen) [reply frontend/demo/component/combobox/combo-box-presentation.ts:7]: I'll create a separate PR for these changes.

File: frontend/demo/fusion/authentication/auth.ts (hunk 9 above)
Comment (vursen) [reply frontend/demo/fusion/authentication/auth.ts:10]: Good catch, fixed.

File: frontend/demo/foundation/material-tokens.ts (hunk 8 above)
Comment (vursen) [reply frontend/demo/foundation/material-tokens.ts:7]: Good catch, fixed.

File: frontend/demo/foundation/lumo-tokens.ts (hunk 7 above)
Comment (vursen) [reply frontend/demo/foundation/lumo-tokens.ts:18]: Good catch, fixed.

File: frontend/demo/fusion/forms/field-strategy/my-binder.ts (hunk 10 above)
Comment (vursen) [reply frontend/demo/fusion/forms/field-strategy/my-binder.ts:1]: I'll create a separate PR for these changes.

**Comment by vursen** [comment]
Closing in favor of #3714

**Pull request closed without merging**